### ISSUE TITLE
json encode error : TypeError: Decimal('100000000000000000000') is not JSON serializable when sending a transaction

### ISSUE THREAD

**Issue by euri10** [pipermerriam/web3.py#58]
* Version: 1.6.0
* Python: 2.7/3.4/3.5
* OS: linux

### What was wrong?
I'm trying to create an account and send funds to it from the coinbase I got this error

```
DEBUG:requests.packages.urllib3.connectionpool:"POST / HTTP/1.1" 200 80
E
web3 = <web3.main.Web3 object at 0x7fc57de45d30>
request = <SubRequest 'account' for <Function 'test_buyAsset[rpc-0-param1]'>>
wait_for_transaction = <function wait_for_transaction.<locals>._wait_for_transaction at 0x7fc57e6a7d08>

    @pytest.fixture(params=[0])
    def account(web3, request, wait_for_transaction):
        if (len(web3.personal.listAccounts))< 2:
            log.debug('web3: {} {}'.format(web3, web3.personal.listAccounts))
            new_account = create_account(web3, password='test')
            log_who_got_what(web3)
>           tx = web3.eth.sendTransaction({"from": web3.eth.coinbase, "to": new_account, "value": to_wei(100, 'ether')})

PycharmProjects/ddn/tests/test_dudes.py:43: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.5/dist-packages/web3-1.6.0-py3.5.egg/web3/eth.py:208: in sendTransaction
    [formatted_transaction],
/usr/local/lib/python3.5/dist-packages/web3-1.6.0-py3.5.egg/web3/providers/manager.py:20: in request_blocking
    response_raw = self.provider.make_request(method, params)
/usr/local/lib/python3.5/dist-packages/web3-1.6.0-py3.5.egg/web3/providers/rpc.py:17: in make_request
    request = self.encode_rpc_request(method, params)
/usr/local/lib/python3.5/dist-packages/web3-1.6.0-py3.5.egg/web3/providers/base.py:24: in encode_rpc_request
    "id": next(self.request_counter),
/usr/lib/python3.5/json/__init__.py:230: in dumps
    return _default_encoder.encode(obj)
/usr/lib/python3.5/json/encoder.py:198: in encode
    chunks = self.iterencode(o, _one_shot=True)
/usr/lib/python3.5/json/encoder.py:256: in iterencode
    return _iterencode(o, 0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.encoder.JSONEncoder object at 0x7fc5833e9f98>
o = Decimal('100000000000000000000')

    def default(self, o):
        """Implement this method in a subclass such that it returns
            a serializable object for ``o``, or calls the base implementation
            (to raise a ``TypeError``).
    
            For example, to support arbitrary iterators, you could
            implement default like this::
    
                def default(self, o):
                    try:
                        iterable = iter(o)
                    except TypeError:
                        pass
                    else:
                        return list(iterable)
                    # Let the base class default method raise the TypeError
                    return JSONEncoder.default(self, o)
    
            """
>       raise TypeError(repr(o) + " is not JSON serializable")
E       TypeError: Decimal('100000000000000000000') is not JSON serializable

/usr/lib/python3.5/json/encoder.py:179: TypeError
```

#### Cute Animal Picture

![parresseux](http://pmdvod.nationalgeographic.com/NG_Video/742/275/140715-baby-sloth-orphanage-rescue-vin_640x360_304699971673.jpg)

**Comment by euri10** [comment]
fixed by casting the to_wei Decimal object to int
`tx = web3.eth.sendTransaction({"from": web3.eth.coinbase, "to": new_account, "value": int(to_wei(100, 'ether'))})`
leaving the issue opened, i don't know if it's a "bug" or if I'm supposed to pass int

**Comment by pipermerriam** [comment]
This is a bug.

**Comment by pipermerriam** [comment]
The utility functions need to be modified to cast return values to integers.

**Comment by pipermerriam** [comment]
Fixed via #66
